bin/docker-push

```diff
@@ -41,8 +49,9 @@ if [ "$DOCKER_BUILD_CACHE_FROM" == "available" ]; then
   done <<< "$(grep -i '^FROM.* AS ' ${DOCKERFILE} | awk '{print $4}')"
 fi
 
-echo "Pushing ${DOCKERTAG}:latest as ${CI_SHA1}, ${CI_REF}, ${CI_BUILD_NUM} to ${EXTERNAL_REGISTRY_BASE_DOMAIN}/${REPOSITORY_NAME}"
-
-tag_push "${DOCKERTAG}:latest" "${EXTERNAL_REGISTRY_BASE_DOMAIN}/${REPOSITORY_NAME}:${CI_SHA1}"
-tag_push "${DOCKERTAG}:latest" "${EXTERNAL_REGISTRY_BASE_DOMAIN}/${REPOSITORY_NAME}:${CI_REF}"
-tag_push "${DOCKERTAG}:latest" "${EXTERNAL_REGISTRY_BASE_DOMAIN}/${REPOSITORY_NAME}:build_${CI_BUILD_NUM}"
+for TAG_VERSION in $(echo "$TAG_VERSIONS" | tr "$TAG_VERSION_DELIMITER_CHAR" ' '); do
```

Comment:
I like the extensibility here! Although I generally avoid the use of arrays in bash, I think they make sense here due to how we use `*.config` files for pipeline configurations. I'll give an example below:

The `development.config` below shows an example configuration with your current implementation.
```bash
# development.config - the pipeline parameters for rok8s-scripts development pipeline jobs
NAMESPACE='development'
SOPS_SECRETS=('development/some-secret-file')
HELM_CHARTS=('charts/myapp')
HELM_RELEASE_NAMES=('myapp-dev')
HELM_VALUES=('development/myapp')

# EXAMPLE IMPLEMENTATION FOR OVERRIDE
LATEST=something-not-latest
SOME_ADDTNL_TAG=another-tag # this could have been reduced into the line below but I wanted to be consistent with env var reference
TAG_VERSIONS="$CI_SHA1$TAG_VERSION_DELIMITER_CHAR$CI_REF$TAG_VERSION_DELIMITER_CHAR$BUILD_TAG_PREFIX$CI_BUILD_NUM$TAG_VERSION_DELIMITER_CHAR$SOME_ADDTNL_TAG"
```

Since we're already using bash arrays in configurations for other downstream scripts, we could change the workflow to have a default set of tags that are built and be able to provide additional tags if wanted, OR override the default tags. See the config example below.

```bash
# development.config - the pipeline parameters for rok8s-scripts development pipeline jobs
NAMESPACE='development'
SOPS_SECRETS=('development/some-secret-file')
HELM_CHARTS=('charts/myapp')
HELM_RELEASE_NAMES=('myapp-dev')
HELM_VALUES=('development/myapp')

# DIFFERENT EXAMPLE IMPLEMENTATION FOR OVERRIDE
DOCKER_LATEST_TAG=something-not-latest
ADDITIONAL_DOCKER_PUSH_TAGS=('another-tag' 'yet-another')
```

The above example give the user the ability to add tags without having to restate default tags they may still want. The script implementation can still have `TAG_VERSIONS` but instead of a strict override we could say `TAG_VERSIONS+=(${ADDITIONAL_DOCKER_PUSH_TAGS[@]})` to append the array of additional tags (if defined). 

I'm definitely open to a conversation here and my intent is not to block but provide a different user experience for people defining parameters in the `*.config` files. If you want to chat on a video call we can also talk through this too!

Reply:
That's a pretty easy fix too, although I'd personally rather folks either take the default tags or provide their own.  Seems overkill to me to allow them to specify additional tags.  Once they want to take control, then they should just take control.  A first impl I'll push will be to use a bash array, then you can consider whether we should allow _additional_ tags to be specified versus just allowing _tags_ to be specified.

Reply:
BTW:
> LATEST=something-not-latest
SOME_ADDTNL_TAG=another-tag # this could have been reduced into the line below but I wanted to be consistent with env var reference
TAG_VERSIONS="$CI_SHA1$TAG_VERSION_DELIMITER_CHAR$CI_REF$TAG_VERSION_DELIMITER_CHAR$BUILD_TAG_PREFIX$CI_BUILD_NUM$TAG_VERSION_DELIMITER_CHAR$SOME_ADDTNL_TAG"

If you notice, `TAG_VERSION_DELIMITER_CHAR` defaults to `,`, so I think the above would become
```
LATEST=something-not-latest
SOME_ADDTNL_TAG=another-tag # this could have been reduced into the line below but I wanted to be consistent with env var reference
TAG_VERSIONS="$CI_SHA1,$CI_REF,$BUILD_TAG_PREFIX$CI_BUILD_NUM,$SOME_ADDTNL_TAG"
```
I didn't really want to make the tag delimiter be configurable lest it be overkill, but it _is_ the more general solution, so I went with that; I didn't want someone to come along later & say, "Hey, _I_ like to delimit with colons"...

Reply:
Oh, hell.  I went ahead & just added support for specifying _additional_ tags, too.  Also changed `TAG_VERSIONS` to `DOCKER_TAG_VERSIONS` to be consistent.  The script now expects `DOCKER_TAG_VERSIONS` and `ADDITIONAL_DOCKER_TAG_VERSIONS` to be arrays.